src/Chapter2.hs

```diff
@@ -605,7 +610,9 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [h] = [h, h]
```

Comment:
This case is redundant. Pattern matching on `[h]` is equivalent to `h : []` and the next lines handle this case pretty well :ok_hand: